README.md

```diff
@@ -29,23 +29,145 @@ This proposal reached Stage 1 at the 2020 Feb TC39 meeting.
+  |  Dotted       |1: 50: 00              |
+
+
+#### Design
+  * The user can determine the formatting width using a  parameter `style` and the value of this parameter take a ` string` as following:
+      * “wide” 
+      * “short”
+      * “narrow”
+      * “numeric” 
+      * “dotted”
+
+### Supported Time Fields
+Users need the following fields to be supported
+  * Year
+  * Month
+  * Week
+  * Day
+  * Hour
+  * Minute
+  * Second
+  * Millisecond
+  * Microsecond
+  * Nano Second
+
+#### Design
+We are going to support the following fields as a beginning
+  * Week
+  * Day
+  * Hour
+  * Minute
+  * Second
+  * Millisecond
+  * Microsecond
+  * Nano Second
+
+### Determining Time Fields
+Users need to determine the time fields,
+ * For example: 6059 seconds
+   * Hours, Minutes, Seconds
+     * 1 hours, 40 minutes and  59 seconds
+   * Minutes and Seconds
+     * 100 minutes and 59 seconds.
+ 
+#### Design
+   * Users can determine the time fields using `LargestField` and `SmallestField` parameters.
+   * Example: 
+   ```javascript
+    {
+      LargestField: "hour",
+      SmallestField: "second"
+    }
+  ```
+   * Default values
+    ```javascript
+         LargestField: "Day",
+         SmallestField: "second"
+    ```
+### Hide zero-value fields
+Users needs to hide zero-value fields depends in some criteria
+  * Hide all zero-value fields
+    * Hours, Minutes, Seconds (without the `hide` option)
+      * 1 hours, 0 minutes and 59 seconds.
+    * Hours, Minutes, Seconds (with the `hide` option)
+      * 1 hours and 59 seconds.
+  * Hide-upper-limit with zero-valued.
+    * Hours, Minutes, Seconds, MilliSeconds (without the `hide-upper-limit` option)
+      * 0 hr, 20 min, 0 sec, 100 msec .
+    * Hours, Minutes, Seconds MilliSeconds(with the `hide-upper-limit` option)
+      * 20 min, 0 sec, 100 msec.
+  * Hide-lower-limit with zero-valued.
+
+#### Design
+- Users can set the `HidingZeroValuedOption` parameter  by one of the following values:
+    * “ALL”, // Hide all the fields that have a zero-valued.
+    * “LEADING_AND_TRAILING”, // Hide all the zero fields in the leading or the training
+    * “LEADING_ONLY”,
+    * “TRAILING_ONLY”, 
+    * “NONE”, // Do not hide any zero-valued fields.
+
+- Default Value: “NONE”
+
+### Locale aware format
+ - Users needs the formatting to be dependent on the locale
+ - For example:
+    * en-US
+      * 1 hour, 46 minutes and 40 seconds
+    * fr-FR
+      * 1 heure, 46 minutes et 40 secondes
 
 ## API design
+I propose to have a class called `DurationFormat` 
 
-  ```javascript
-    const formatter =  new Intl.DurationFormatter('en-US' , { style: 'short', fields: 'hours-minutes-seconds'});
 
-    const duration =  Temporal.Duration.from({
-                        hours: 2, minutes: 46, seconds: 40
-                      });
+  ```javascript
+    const formatter =        
+        new Intl.DurationFormat({
+                              locale: 'en-US',
```

Comment:
The locale is always the first argument.  Also, your casing conventions are not correct.

I think you would want:

```javascript
const formatter = new Intl.DurationFormat('en-US', {
    largestField: 'hour',
    smallestField: 'second', 
		style: 'short',
		round: true,
		hidingZeroValuedOption: NONE,
});
```

As stated before, I'm not 100% onboard with this API, but it's good enough for Stage 2.

Reply:
+1 on locale as the first argument. Regarding the options, I thought we had decided to use the more explicit `"fields"` option for Stage 2 and iterate on these solutions before Stage 3, so that's what I did in the spec.

Reply:
Done !